install.sh

```diff
@@ -1,62 +1,167 @@
+    exit 1
+}
+
+install_debian () {
+    sudo apt install python3 python3-tk python3-pip git || install_fail
+}
+
+install_fedora () {
+    sudo dnf install python3 python3-tkinter python3-pip git || install_fail
+}
+
+install_centos () {
+    sudo yum install -y python3 || install_fail
+    rpm -q epel-release &> /dev/null || EPEL=0
+    sudo yum install -y python3-tkinter epel-release python3-pip git || install_fail
+    # Honestly not sure why it's installing epel and then uninstalling
+    [[ $EPEL -eq 0 ]] && sudo yum remove -y epel-release
+}
+
+install_arch () {
+    sudo pacman -S --needed python3 tk git && python3 -m ensurepip || install_fail
+}
+
+install_macos() {
+    echo "This script will install homebrew, if you do not wish to install homebrew, exit within 5 seconds..."
+    sleep 5
+    if test ! $(which brew); then
+        #Install homebrew
+        echo "Installing homebrew..."
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+    fi
+    # Install python3.10 & git or fail out
+    brew install python@3.10 && \
+    brew install git || \
+        install_fail
+}
+
+install_python_requirements () {
+    python3 -m pip install -q -r requirements.txt --user && \
+        echo "Python requirements installed." || exit 1
+    # Pip should give an error if it fails.
+}
+
+install_os_requirements () {
+    echo "YT-Spammer-Purge has a few requirements that you will need to install."
+    sleep 1
+
+
+    # Check for known OS's
+    INSTALLED=0
+
+    case "$(uname -s)" in
+        Darwin*) install_macos && INSTALLED=1 || install_fail;;
+        # Mac
+        #CYGWIN*) do_something;;
+        # Cygwin Emulator -- Unimplemented
+        #MINGW*) do_something;;
+        # MINGW Emulator -- Unimplemented
+    esac
+
+    [[ -e /etc/debian_version ]] && install_debian && INSTALLED=1
+
+    [[ -e /etc/fedora-release ]] && install_fedora && INSTALLED=1
+
+    [[ -e  /etc/centos-release ]] && install_centos && INSTALLED=1
+
+    [[ -e /etc/arch-release ]] && install_arch && INSTALLED=1
+
+    [[ $INSTALLED -eq 0 ]] && printf "You are on an unknown system. You will have to install the required packages manually.\nContributions are welcome to add support for your system:\nhttps://github.com/ThioJoe/YT-Spammer-Purge" && exit 1
 
-TAG=$(curl https://api.github.com/repos/ThioJoe/YT-Spammer-Purge/releases/latest -s | jq .name -r)
-if  [[ $JQ -ne 0 ]]; then
-	if [[ -e /etc/debian_version ]]; then
-		sudo apt purge jq
-	elif [[ -e /etc/fedora-release ]]; then
-		sudo dnf remove jq
-	elif [[ -e /etc/centos-release ]]; then
-		sudo yum remove jq
-	elif [[ -e /etc/arch-release ]]; then
-		sudo pacman -Rs jq
-	fi
-else
-	echo "Did not uninstall jq as it was preinstalled before running this script."
+}
+
+install_latest_release () {
+    echo "We are now going to download the code for YT-Spammer-Purge."
+    sleep 1
+    git clone https://github.com/ThioJoe/YT-Spammer-Purge
+    cd YT-Spammer-Purge
```

Comment:
Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```suggestion
    cd YT-Spammer-Purge || exit
```

See for more details: https://github.com/koalaman/shellcheck/wiki/SC2164

Reply:
I didn't even consider this as a possibility 😮
Good catch :+1: